### PR TITLE
Memory savings

### PR DESCRIPTION
**Pull request by ssanderson**

### CODE REVIEW

**Comment by ssanderson** [comment]
These are all relatively minor in the grand scheme of things.  The overwhelming consumer of memory in zipline simulations that touch many assets are the active blocks of minutely bcolz files.

File: zipline/data/history_loader.py [hunk 1]
```diff
@@ -380,7 +380,7 @@ def _ensure_sliding_windows(self, assets, dts, field,
         assets = self._asset_finder.retrieve_all(assets)
 
         try:
-            end_ix = self._calendar.get_loc(end)
+            end_ix = self._calendar.searchsorted(end)
```
Comment (ehebert) [inline zipline/data/history_loader.py]: Thanks for spotting this one. I was unaware of that bug/behavior.

On the same track, I do question whether we should pull in all minutes ever from the calendar (mea culpa), in favor of calculating the location in a similar fashion to the minute bar reader's find position of minute.
(The minute index takes 11MB at minimum for 14 years just for the datetime values.)

Does using `getitem` also trigger the creation of the hash table?
If so we should also audit usage of `trading_calendar.previous_minute`.

File: zipline/utils/cache.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ class Expired(Exception):
     """
 
 
-class CachedObject(namedtuple("_CachedObject", "value expires")):
```
Comment (ehebert) [inline zipline/utils/cache.py]: Why the change from using `namedtuple`.

Comment (ssanderson) [inline zipline/utils/cache.py]: @llllllllll and I had some concerns about the behavior of garbage collection on objects with empty slots.  See the discussion in https://bugs.python.org/issue24379.

The specific issue, I think, is that objects with empty `__slots__` don't participate in garbage collection, which means they can leak if they participate in a cycle.  I don't think that's happening here, but there's also not much benefit in making this a subclass of namedtuple.
